bitnami/opensearch/README.md

```diff
@@ -342,6 +342,7 @@ You can enable this initContainer by setting `volumePermissions.enabled` to `tru
 | `master.nameOverride`                                      | String to partially override opensearch.master.fullname                                                                                                                                                                         | `""`                |
 | `master.fullnameOverride`                                  | String to fully override opensearch.master.fullname                                                                                                                                                                             | `""`                |
 | `master.servicenameOverride`                               | String to fully override opensearch.master.servicename                                                                                                                                                                          | `""`                |
+| `master.service.trafficDistribution`                       | String Traffic distribution for the master service                                                                                                                                                                              | `PreferClose`       |
```

Comment:
I think I didn't explain myself correctly. I was talking about including it under the existing `service` dict:

- https://github.com/bitnami/charts/blob/main/bitnami/opensearch/values.yaml#L354-L358

Is there a reason why you want to customize this per component? E.g. setting `PreferClose` for "master" nodes while using the default routing strategy for the rest.

Reply:
In my opinion, the default value should be set to 'PreferClose' for all services for various reasons (performance, cost, reliability), especially for master and coordinating nodes. However, depending on the use cases, it might be necessary to change this setting on a per-component basis when pods on the same node or zone become overloaded.

Another option is to establish a global 'service.trafficDistribution' parameter that applies to all components while still allowing for component-specific overrides, such as 'master.service.trafficDistribution', 'data.service.trafficDistribution', 'coordinating.service.trafficDistribution', and 'ingest.service.trafficDistribution'.

What do you think?

Reply:
I see what you mean. Given we want to change setting on a per-component basis, I created this: https://github.com/bitnami/charts/pull/32507

Once it's merged (it should be merged today), please rebase your local branch to have latest changes in `main` and add this new parameter as part of `xxx.service.headless` objects.

Reply:
Thanks, I noticed your pull request was just merged. I'll take care of the rebase and the modifications, and I'll keep you updated.

Reply:
Done :heavy_check_mark: